### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by VladikVll**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,27 @@
...
+        "status": "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(report, linter_report[report])
+        for report in linter_report
```
Comment (Dimosphen1) [inline app/main.py]: Use `linter_report.items()` and perform unpacking to avoid retrieving value by key

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,27 @@
...
-    pass
+    return {
+        "errors": [],
+        "path": file_path,
+        "status": "passed"
+    } if not errors else {
+        "errors": [format_linter_error(err) for err in errors],
+        "path": file_path,
+        "status": "failed"
+    }
```
Comment (Dimosphen1) [inline app/main.py]: It would be beneficial not to duplicate code and add value to only the `status` key using the if condition.

**Review by Dimosphen1 — CHANGES_REQUESTED** [review]
Several changes were requested

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,23 @@
...
+        "status": "passed" if not errors else "failed"
+    }
 
 
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+    return [
+        format_single_linter_file(key, value)
+        for key, value in linter_report.items()
```
Comment (vsmutok) [inline app/main.py]: `key`, `value` not clear variable  name

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/main.py (hunk 3 above)
Comment (VladikVll) [reply app/main.py]: Дякую))

**Review by viktoria-rybenchuk — APPROVED** [review]
Good job!

**Pull request closed without merging**